### ISSUE TITLE
Statistical equivalence test suite against sklearn reference values

### ISSUE THREAD

**Issue by dollspace-gay** [dollspace-gay/ferrolearn#9]
## Summary

Create a comprehensive statistical equivalence test suite that proves ferrolearn produces numerically identical (within ULP budget) results to scikit-learn across all algorithms, serving as empirical mathematical proof of correctness.

## Context

The design doc (Section 20.3) requires statistical equivalence benchmarking. Oracle tests currently exist for individual algorithms, but there is no systematic framework that:
- Tests against sklearn on a standardized battery of datasets
- Measures maximum ULP divergence per algorithm
- Produces an auditable equivalence report
- Covers edge cases (near-singular matrices, highly correlated features, class imbalance)

This is the empirical complement to Kani's formal proofs — Kani proves properties hold for all inputs within bounds, this proves numerical agreement with the reference implementation on realistic data.

### Test matrix

Each algorithm tested against sklearn on:
- **Toy datasets**: iris, digits, boston-like, make_classification, make_regression
- **Edge cases**: single feature, single sample (where valid), n_samples < n_features, perfect collinearity, constant features, all-same-class
- **Numerical stress**: very large values (1e15), very small values (1e-15), mixed scales
- **Dtypes**: f32 and f64

### Equivalence criteria

| Component | Tolerance |
|---|---|
| Predictions (regression) | max relative error < 1e-10 (f64), < 1e-5 (f32) |
| Predictions (classification) | Exact label match on all test datasets |
| Probabilities | max absolute error < 1e-10 (f64) |
| Coefficients / weights | max relative error < 1e-8 (f64) |
| Metric scores | max absolute error < 1e-12 (f64) |
| Transform outputs (PCA, Scaler) | max relative error < 1e-10 (f64) |

### Algorithms to cover

**Regressors**: LinearRegression, Ridge, Lasso, ElasticNet
**Classifiers**: LogisticRegression, DecisionTreeClassifier, RandomForestClassifier, KNeighborsClassifier, GaussianNB
**Transformers**: StandardScaler, PCA
**Clusterers**: KMeans
**All 17 metrics**: classification, regression, clustering

### Implementation

Python script generates reference values from sklearn, serializes to JSON fixtures. Rust tests load fixtures and compare. This is an extension of the existing oracle test pattern but systematized.

```
tests/
  fixtures/
    sklearn_reference/
      linear_regression_iris.json     # {X, y, coef, intercept, predictions, score}
      linear_regression_edge_singular.json
      ...
  statistical_equivalence/
    mod.rs
    test_regressors.rs
    test_classifiers.rs
    test_transformers.rs
    test_metrics.rs
```

### Report generation

A CI job produces an equivalence report:
```
Algorithm Equivalence Report
============================
LinearRegression    coef max_rel_err=2.3e-14  pred max_rel_err=1.1e-14  PASS
Ridge               coef max_rel_err=5.7e-13  pred max_rel_err=3.2e-13  PASS
...
accuracy_score      max_abs_err=0.0                                      PASS
f1_score            max_abs_err=0.0                                      PASS
```

## Acceptance criteria
- [ ] Reference fixture generation script covering all 12 models + 17 metrics
- [ ] At least 5 datasets per algorithm (including 2+ edge cases)
- [ ] All equivalence tests pass within stated tolerances
- [ ] CI produces equivalence report on every PR
- [ ] Any divergences found are documented with root cause analysis (different algorithm variant, floating point ordering, etc.)
